runtime/doc/api.txt

```diff
@@ -976,6 +976,12 @@ nvim_get_hl_id_by_name({name})                      *nvim_get_hl_id_by_name()*
 
     similar to |hlID()|, but allocates a new ID if not present.
 
+nvim_get_hl_ns()                                            *nvim_get_hl_ns()*
```

Comment:
This can take `opt` arg which optionally accepts a `win` key. Don't need the `nvim_win_*` variant.

This also seems to be another case where we didn't future-proof `nvim_get_namespaces()`.

Should this instead be named `nvim_get_ns()` and take an `opt` that controls the results? Why is "highlight" namespaces special enough to need a dedicated function?

```suggestion
nvim_get_ns({opt})                                            *nvim_get_ns()*

Parameters:
  {opt} Dict with optional keys:
    - "hl": include highlight namespaces
    - "win": only get window-local namespaces for this window-id
```

Reply:
>  This would be inconsistent with `nvim_win_set_hl_ns` and `nvim_set_hl_ns`,

We'll fix that eventually. We need to stop the `nvim_buf` and `nvim_win` stuff, it was a mistake.

> There is a slight difference in the permitted return values. In the case for windows, the function can return -1, as that is the default value and is permitted by nvim_win_set_hl_ns. This is not the case for nvim_set_hl_ns.

If the caller passes `opt.win`, they should expect the behavior they opted-into.

> based on the text under :help api-contract, I believe it's too late to merge nvim_win_set_hl_ns and nvim_set_hl_ns so that it would be consistent with a single get function.

Yes, but we could introduce `nvim_set_ns`.

But I'm not certain it makes sense for "highlight namespaces" to be grouped with "general namespaces". Need others to comment on that.

Reply:
I don't understand why highlight namespaces exist at all. Couldn't we just use regular namespaces for highlights? If so perhaps we can add retrofit this somehow and deprecate window namespaces?

Reply:
> "This can take opt arg which optionally accepts a win key. Don't need the nvim_win_* variant."

I've updated to code to now have a single function `nvim_get_hl_ns` that optionally accepts a `winid`.

Reply:
> I don't understand why highlight namespaces exist at all.

For reference: https://github.com/neovim/neovim/pull/25455#discussion_r1345851839